api/src/clojure/org/akvo/flow_api/unilog/unilog.clj

```diff
@@ -154,22 +156,25 @@
     (let [ds (DatastoreServiceFactory/getDatastoreService)
           events (process-new-events (read-event-log db offset))
           user-id (user/id-by-email-or-throw-error remote-api instance-id email)
-          authorized-forms (set (map ds/id (su/list-forms user-id)))
-          authorized-forms-to-load (set/intersection authorized-forms (:forms-to-load events))
+          authorized-forms (set (map ds/id (su/list-forms user-id (:forms-to-load events))))
           form-id->form (reduce (fn [acc form-id]
                                   (assoc acc form-id (su/get-form-definition (long form-id)
                                                                              {:include-survey-id? true})))
                                 {}
-                                authorized-forms-to-load)
-          authorized-surveys (set (map ds/id (su/list* user-id)))
-          authorized-events (filter-events-by-authorization events form-id->form authorized-forms authorized-surveys)
+                                authorized-forms)
+          authorized-surveys (su/list* user-id (:survey-changed events))
+          survey-id->survey (reduce (fn [acc survey]
+                                      (assoc acc (ds/id survey) (su/survey->map survey)))
+                                    {}
+                                    authorized-surveys)
+          authorized-events (filter-events-by-authorization events form-id->form survey-id->survey)
           form-instances (doall
                            (mapcat
                              (fn [to-load]
                                (form-instance/by-ids ds (:form to-load) (:form-instance-ids to-load)))
                              (:form-instances-to-load authorized-events)))
-          data-point-changed (doall (data-point/by-ids ds (map :id (:data-point-changed authorized-events))))
-          survey-changed (doall (su/by-ids ds (:survey-changed authorized-events)))]
+          data-point-changed (doall (data-point/by-ids ds (:data-point-changed authorized-events)))
+          survey-changed (vals survey-id->survey)]
```

Comment:
Should use the result of `filter-events-by-authorization`, instead of doing the calculation here again